### PR TITLE
Indentation check refactoring

### PR DESCRIPTION
**Pull request by VelinYordanov**
Added exceptions property for the indentation check and made it more abstract.

Signed-off-by: VelinYordanov <velin.iordanov@gmail.com>

### CODE REVIEW

File: custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/IndentationCheck.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    
+    public void setFileExtensionsToIndentation(String value) {
+        String[] fileExtensionsAndIndentation = value.split("[-,]");
+        int length = fileExtensionsAndIndentation.length;
+        if(length % 2 != 0) {
+            logger.error("FileExtensionsToIndentation not set correctly. It should be file extensions and indentation character.Skipping last parameter.");
+            length-=1;
+        }
+        
+        for(int i =0;i<length;i+=2) {
```
Comment (martinvw) [inline custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/IndentationCheck.java]: Formatting looks of

File: custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/IndentationCheck.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+ * @author Velin Yordanov - Made the check more generic and added exceptions property
+ *
+ */
+public class IndentationCheck extends AbstractStaticCheck {
+    private static final String TAB_CHARACTER = "\t";
+    private static final String SPACE_CHARACTER = " ";
+    private static final String TABS = "tabs";
+    private static final String WARNING_MESSAGE = "Not supported indentation characters used";
+    private boolean onlyShowFirstWarning;
+    private Map<String, String> fileExtensionsToIndentation =  new Hashtable<String, String>();
```
Comment (martinvw) [inline custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/IndentationCheck.java]: Please use the diamond operator, ie replace the second occurrence of the same generic by `<>`

Comment (wborn) [reply custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/IndentationCheck.java]: There is a [DiamondOperatorForVariableDefinitionCheck](https://sevntu-checkstyle.github.io/sevntu.checkstyle/apidocs/com/github/sevntu/checkstyle/checks/coding/DiamondOperatorForVariableDefinitionCheck.html) for this in [Sevntu-Checkstyle](http://sevntu-checkstyle.github.io/sevntu.checkstyle/). :smile:  
Did anyone ever have a look at and consider using any of those checks?

**Comment by kaikreuzer** [comment]
Closing due to inactivity.

-

**Pull request closed without merging**